src/main/java/com/dropwizard/example/h2/resources/PersonResource.java

```diff
@@ -0,0 +1,97 @@
+package com.dropwizard.example.h2.resources;
+
+import com.codahale.metrics.annotation.Timed;
+import com.dropwizard.example.h2.DAO.PersonDAO;
+import com.dropwizard.example.h2.model.Person;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Optional;
+
+@Path("/persons")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class PersonResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PersonResource.class);
+
+    private final PersonDAO personDAO;
+
+    public PersonResource(PersonDAO personDAO) {
+        this.personDAO = personDAO;
+    }
+
+    @GET
+    @Timed
+    public Response getAllPersons() {
+        List<Person> persons = personDAO.getAll();
+        LOGGER.info("Fetched persons count - " + persons.size());
+
+        return Response.ok(persons).build();
+    }
+
+    @GET
+    @Timed
+    @Path("{id}")
+    public Response getPersonById(@PathParam("id") Long id) {
+
+        LOGGER.info("Fetching person by id {} ", id);
+        Optional<Person> personFromDb = personDAO.getById(id);
+        if (personFromDb.isPresent()) {
```

Comment:
Try to rewrite it more in functional style.

### **Hint**. orElseGet()

- [ ] Rewrite in functional style